ArrayLimitedQueue/ArrayLimitedQueue.swift

```diff
@@ -0,0 +1,173 @@
+//
+//  ArrayLimitedQueue.swift
+//
+//  Created by Sergey Pugach on 15.04.17.
+//  Copyright © 2017 Sergey Pugach. All rights reserved.
+//
+
+/*
+ Array Limited Queue is a collection that has the features
+ of a regular array (only get + limited size), queues and sets.
+ The way you value and compare items can be user-defined.
+ */
+
+import Foundation
+
+public struct ArrayLimitedQueue<T: Comparable> {
+    
+    //The maximum number of items in the collection. Can be changed after
+    public var maxStoredItems: Int = 1 {
+        didSet {
+            let sizeDiff = internalArray.count - maxStoredItems
+            if 0 < sizeDiff && sizeDiff <= internalArray.count {
+                internalArray.removeFirst(sizeDiff)
+            }
+        }
+    }
+    /*
+     This property determines which elements can be in the collection.
+     After the change from true to false. Negative values will be deleted
+     Note: When using custom types, you need to set the zeroValue.
+     */
+    public var zeroValue:T? = 0 as? T
+    public var positiveValues: Bool = false {
+        didSet {
+            if positiveValues {
+                
+                if let zero = zeroValue {
+                    internalArray = internalArray.filter{$0 > zero}.map{ $0 }
+                    
+                } else {
+                    fatalError("A zeroValue is not setted")
+                }
+            }
+        }
+    }
+    
+    /*
+     Specifies whether to remove duplicates from the collection.
+     When a copy is found, it leaves the last value.
+     */
+    public var deleteExisting = true {
+        didSet {
+            if deleteExisting {
+                for i in stride(from: internalArray.count - 1, to: 0, by: -1) {
+                    
+                    for j in stride(from: i - 1, to: 0, by: -1) {
+                        
+                        if internalArray[i] == internalArray[j] {
+                            print("remove :\(internalArray[j])")
+                            internalArray.remove(at: j)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    func removeDuplicates(accumulator: [T], element: T) -> [T] {
+        return accumulator.contains(element) ? accumulator : accumulator + [element]
+    }
+    
+    public var array: [T] {
```

Comment:
We could probably combine `array` and `internalArray` to a single property, since it seems like the only purpose of having the two is to prevent others from tampering with the internal storage:

```
public private(set) array: [T] = []
```

This will prevent users from manipulating the array directly, but still allow them to read from it.